test/extensions/filters/udp/dns_filter/dns_filter_test.cc

```diff
@@ -2427,6 +2427,133 @@ stat_prefix: "my_prefix"
   EXPECT_EQ(1, config_->stats().known_domain_queries_.value());
 }
 
+TEST_F(DnsFilterTest, WildcardName) {
+  InSequence s;
+
+  const std::string wildcard_virtual_domain = R"EOF(
+stat_prefix: "my_prefix"
+server_config:
+  inline_dns_table:
+    external_retry_count: 0
+    virtual_domains:
+      - name: "*.foobaz.com"
```

Comment:
Can we add some tests for invalid virtual_domain test, like "*.com", "*.", "com", and see whether there is any issue?